dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageLoader.java

```diff
@@ -0,0 +1,212 @@
+            .getTemplate(htmlPage, mode.showLive);
+
+
+        User sys = APILocator.systemUser();
+
+
+        if (cmsTemplate == null || !InodeUtils.isSet(cmsTemplate.getInode())) {
+            Logger.error(This.class, "PAGE DOES NOT HAVE A VALID TEMPLATE (template unpublished?) : page id "
+                    + htmlPage.getIdentifier() + ":" + identifier.getURI());
+        }
+
+
+
+        // List of tags found in this page
+        List<Tag> pageFoundTags = new ArrayList<>();
+
+        // Check if we want to accrue the tags of this HTMLPage contentlet
+        if (Config.getBooleanProperty("ACCRUE_TAGS_IN_PAGES", true)) {
+
+            List<Tag> htmlPageFoundTags = APILocator.getTagAPI()
+                .getTagsByInode(htmlPage.getInode());
+            if (htmlPageFoundTags != null && !htmlPageFoundTags.isEmpty()) {
+                pageFoundTags.addAll(htmlPageFoundTags);
+            }
+        }
+
+
+
+        sb.append("#set($dotPageContent = $dotcontent.find(\"" + htmlPage.getInode() + "\" ))");
+
+
+        // set the host variables
+
+        Host host = APILocator.getHTMLPageAssetAPI()
+            .getParentHost(htmlPage);
+        sb.append("#if(!$doNotParseTemplate)")
+            .append("#parse('")
+            .append(folderPath)
+            .append(host.getIdentifier())
+            .append(".")
+            .append(VelocityType.SITE.fileExtension)
+            .append("')")
+            .append(" #end ");
+
+
+        sb.append(new PageContextBuilder(htmlPage, sys, mode).asString());
+
+
+        // Now we need to use the found tags in order to accrue them each time this page is visited
+        if (!pageFoundTags.isEmpty()) {
+            // Velocity call to accrue tags on each request to this page
+            sb.append("$tags.accrueTags(\"" + TagUtil.tagListToString(pageFoundTags) + "\" )");
+        }
+        if (htmlPage.isHttpsRequired()) {
+            sb.append(" #if(!$ADMIN_MODE  && !$request.isSecure())");
+            sb.append("    #if($request.getQueryString())");
+            sb.append(
+                    "        #set ($REDIRECT_URL = \"https://${request.getServerName()}$request.getAttribute('javax.servlet.forward.request_uri')?$request.getQueryString()\")");
+            sb.append("    #else ");
+            sb.append(
+                    "        #set ($REDIRECT_URL = \"https://${request.getServerName()}$request.getAttribute('javax.servlet.forward.request_uri')\")");
+            sb.append("    #end ");
+            sb.append("    $response.sendRedirect(\"$REDIRECT_URL\")");
+            sb.append(" #end ");
+        }
+
+        sb.append("#if($HTMLPAGE_REDIRECT != \"\")");
+        sb.append(" $response.setStatus(301)");
+        sb.append(" $response.setHeader(\"Location\", \"$HTMLPAGE_REDIRECT\")");
+        sb.append("#end");
+
+
+        sb.append("#if(!$doNotParseTemplate)");
+        if (cmsTemplate.isDrawed()) {// We have a designed template
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "cmsTemplate" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)